### ISSUE TITLE
Show warning when there’s nothing to deploy

### ISSUE THREAD

**Issue by 2opremio** [weaveworks/flux#243]
I spend a while trying to figure out why nothing was being deployed (no commits in service-conf nor bot message) because I was in the wrong service-conf branch and everything there was already deployed.

**Comment by squaremo** [comment]
Yes, we do not give good advice on why things failed, and in part this is because we calculate the plan what you get from `--dry-run` using the running system, but it's actually _executed_ with respect to the files in e.g., `service-conf`.

**Comment by paulbellamy** [comment]
Let's scope this just to "better warning when nothing to do" separate from "how does flux determine what to do"

**Comment by squaremo** [comment]
This is what it looks like (in PR #421 land) when there's nothing to deploy, assuming you're deploying a specific image:

```
$ fluxctl release --all --update-image=quay.io/weaveworks/helloworld:master-a000001
Submitting release job...
Release job submitted, ID d3ca1b25-e275-3ba3-c882-92926ce60d3b
Status: Complete.

Here's what happened:
SERVICE             STATUS   UPDATES
default/helloworld  skipped  image(s) up to date
Took 6.385707s
```

(NB services that don't even use that image aren't shown by default)

> because I was in the wrong service-conf branch and everything there was already deployed.

I'm not sure what this bit means -- were you in a branch other than `master` locally, and expecting to see changes, but `master` did in fact have all the latest images?

**Comment by paulbellamy** [comment]
Fixed in #421
